### ISSUE TITLE
Proposal: dual specular roughness sliders with mix

### ISSUE THREAD

**Issue by sharktacos** [AcademySoftwareFoundation/OpenPBR#210]
Throwing out for discussion the idea of having two specular roughness sliders mixed by a lerp (typically using a mask texture for the mix), similar to Megascans materials in Unreal. 

This can of course be achieved for instance with a ND_mix_float in materialX, but it could be nice to have it built into the material since it is so commonly used by lookdev artists. Thought I'd bring it up for possible consideration.

**Comment by portsmouth** [comment]
Do you mean there should be two separate specular lobes with independent roughness, with a mix weight (called "dual-specular" sometimes)? If so that's being considered, but it would be more a later release as it alters the BSDF implementation somewhat.

**Comment by sharktacos** [comment]
Yes, that is what I was referring to. I was actually imagining a simple linear interpolation mix between two roughness values modulated by a mask that remaps black and white in the mask to the two roughness values, rather than an additional specular lobe. What would be the advantage of having a second specular lobe that changes the BSDF, as opposed to a simple mix for the roughness values within one specular lobe? How would this be differentiated from the coat?

In terms of GUI, one possibility is for it to work similarly to metalness so that the mix defaults to 0 meaning only roughness1 affects the material with roughness2 greyed out until the mix is raised above 0.

Regarding the anisotropy, my thinking is that the artistic purpose is to have subtle variation in the roughness, so I would expect the anisotropy to be coupled. If I wanted to get variation in the anisotropy I would use a tangent map.

**Comment by virtualzavie** [comment]
I understand some models have started exploring this possibility; I think this is available in UE5 for example.
As for OpenPBR, I would lean towards waiting to see if there is an industry consensus over what the future of the specular lobe parametrisation is.

**Comment by portsmouth** [comment]
I think it would be worth prototyping this and doing some renders, to get a sense of how significant the visual difference is.

**Comment by portsmouth** [comment]
>I was actually imagining a simple linear interpolation mix between two roughness values modulated by a mask that remaps black and white in the mask to the two roughness values, rather than an additional specular lobe. 

Ah I see, you meant simply interpolating two roughnesses with a mix weight? I think that would be handled via a node graph most likely, it doesn't really make sense in the shading model itself.

>What would be the advantage of having a second specular lobe that changes the BSDF, as opposed to a simple mix for the roughness values within one specular lobe? How would this be differentiated from the coat?

The coat has an independent NDF, but it also sits on top so generates layering effects (e.g. Fresnel obscures the substrate at grazing).  Whereas a dual-lobe specular on the substrate gives a coat-like control over the highlight, without those layering effects. Apparently it can be useful.

**Comment by sharktacos** [comment]
Yes the dual lobe spec does sound potentially useful.

I agree that the simple mix would be better handled in a node graph.

**Comment by portsmouth** [comment]
@sharktacos Can this be closed then?
